server/src/main/java/org/apache/iotdb/db/query/executor/LastQueryExecutor.java

```diff
@@ -171,8 +175,9 @@ public QueryDataSet execute(QueryContext context, LastQueryPlan lastQueryPlan)
             nonCachedPaths,
             nonCachedDataTypes,
             context.isDebug());
+
     if (nonCachedPaths.isEmpty()) {
-      return resultContainer;
+      return cachedLastPairs.stream().map(pair -> pair.right).collect(Collectors.toList());
```

Comment:
Actually, the reason that we directly return the resultContainer is that we want to reuse it instead of creating another ArrayList, even if the pair.left will never be used. It is for performance. If you really want to not return extra result, you can make readLastPairsFromCache to return Pair<boolean[], List<TimeValuePair>> instead and return res.right.

Reply:
Good point. I refactored the code to solve this problem and make the code more readable. Please take a look.